reactor-core/src/jcstress/java/reactor/core/publisher/StressSubscriber.java

```diff
@@ -35,14 +40,36 @@
 
 	public AtomicBoolean concurrentOnComplete = new AtomicBoolean(false);
 
+	public AtomicBoolean concurrentOnSubscribe = new AtomicBoolean(false);
+
 	public final AtomicInteger onNextCalls = new AtomicInteger();
 
 	public final AtomicInteger onErrorCalls = new AtomicInteger();
 
 	public final AtomicInteger onCompleteCalls = new AtomicInteger();
 
+	public final AtomicInteger onSubscribeCalls = new AtomicInteger();
+
+	public StressSubscriber() {
+		this(Long.MAX_VALUE);
+	}
+	public StressSubscriber(long initRequest) {
+		this.initRequest = initRequest;
+	}
+
+	@Override
+	protected void hookOnSubscribe(Subscription subscription) {
+		if (!guard.compareAndSet(null, Operation.ON_SUBSCRIBE)) {
+			concurrentOnSubscribe.set(true);
+		} else {
+			subscription.request(initRequest);
```

Comment:
shouldn't the request be strictly positive? I see some `new StressSubscriber(0)` usages in the tests

Reply:
0 == no initial request

Reply:
(added javadoc)

Reply:
but there is no `if (initRequest > 0)` check before requesting

Reply:
ah, sorry. good catch indeed